### ISSUE TITLE
Artifact dependency breaks sync

### ISSUE THREAD

**Issue by jceb** [GoogleContainerTools/skaffold#5522]
### Expected behavior

I'd like to define a base image dependency for two artifacts in my skaffold.yaml (as specified here: https://skaffold.dev/docs/pipeline-stages/builders/docker/) and would like to synchronize changed files across from the local folder where I develop.

### Actual behavior

File changes aren't recognized by skaffold.

### Information

- Skaffold version: v1.20.0
- Operating system: Arch Linux torch 5.10.20-1-lts #1 SMP Thu, 04 Mar 2021 12:02:02 +0000 x86_64 GNU/Linux
- Installed via: skaffold.dev
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v2beta12
kind: Config
metadata:
  name: mine
build:
  local:
    concurrency: 0
    push: true
    useBuildkit: true
  artifacts:
    - image: k3d-my-registry.localhost:5111/myreg/base
      docker:
        dockerfile: Dockerfile-base.dev
    - image: k3d-my-registry.localhost:5111/myreg/api
      docker:
        dockerfile: Dockerfile-api.dev
      requires:
      - image: k3d-my-registry.localhost:5111/myreg/base
        alias: BASE
      sync:
        manual:
          - src: ".npmrc"
            dest: "/app"
          - src: "packages.json"
            dest: "/app"
          - src: "lerna.json"
            dest: "/app"
          - src: "yarn.lock"
            dest: "/app"
          - src: "packages/**/*.js"
            dest: "/app"
          - src: "packages/**/package.json"
            dest: "/app"

```

### Steps to reproduce the behavior

1. a clonable repository with the sample skaffold project
2. `skaffold build`
3. `skaffold dev --auto-build=false`
4. Watching for changes
5. make a change to one of the watched files, e.g. `packages/mypkg/package.json`
6.  .. nothing happens, change isn't recognized

**Comment by gsquared94** [comment]
This is identical to https://github.com/GoogleContainerTools/skaffold/issues/5301

**Comment by gsquared94** [comment]
Fixed in #5301
